### PR TITLE
Integrate CodeQL to CI

### LINKED ISSUE #129: Integrate CodeQL to CI
### Description

To maintain high code quality, Lisk Service's CI pipeline must integrate a code analysis tool such as CodeQL.

### Acceptance Criteria

- Integrate CodeQL to the CI pipeline
- The following CodeQL rule are being executed
  - js/redos

### PR DESCRIPTION
**Pull request by priojeetpriyom**
### What was the problem?

This PR resolves #129 

### How was it solved?

- [x] Add codeql coverage in github action
- [x] Add dependency check in github action 

### How was it tested?
- [x] ci

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: true
+      matrix:
+        language: ['typescript']
```
Comment (sameersubudhi) [inline .github/workflows/codeql.yml:35]:
```suggestion
        language: ['javascript']
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (priojeetpriyom) [reply .github/workflows/codeql.yml:35]: Discussed offline.

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+# supported CodeQL languages.
+#
+name: 'CodeQL'
+
+on:
+  push:
+    branches: ['development', feature/*, hotfix/*, main, release/*]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: ['development', feature/*, hotfix/*, main, release/*]
```
Comment (nagdahimanshu) [inline .github/workflows/codeql.yml:19]: @sameersubudhi Just wanted to confirm, should we also remove `main` branch similar to `lisk-core`

Comment (sameersubudhi) [reply .github/workflows/codeql.yml:19]: Yes please 👍

Comment (sameersubudhi) [inline .github/workflows/codeql.yml:19]:
```suggestion
    branches: ['development', feature/*, hotfix/*, release/*]
```

**Review by nagdahimanshu — APPROVED** [review]

**Pull request merged**